### PR TITLE
feat(notifications): notification basics

### PR DESCRIPTION
**Pull request by Jekrimo**
**What this PR does** 📖

Set up basics for notifications implementation

**Which issue(s) this PR fixes** 🔨

AP-1981

### CODE REVIEW

**Review by aewing — APPROVED** [review]

**Review by stavares843 — CHANGES_REQUESTED** [review]
pnpm-lock.yaml file 🔨 

image

**Review by stavares843 — COMMENTED** [review]
Captura de ecrã 2022-08-16, às 20 22 10


this is missing some linting

you can use `npx prettier --write .` 🔨

**Comment by Jekrimo** [comment]
> pnpm-lock.yaml file 🔨
> 
> image

I didn't mess with that file, should I just delete the file altogether you think?

**Comment by stavares843** [comment]
notifications work 🎉 

Captura de ecrã 2022-08-16, às 20 20 17
Captura de ecrã 2022-08-16, às 20 20 48

File: components/views/navigation/toolbar/alerts/Alerts.vue [hunk 1]
```diff
@@ -1,21 +1,31 @@
...
 <script lang="ts">
-import Vue from 'vue'
-import { mapGetters, mapState } from 'vuex'
+import Vue, {PropType} from 'vue'
+import { mapState } from 'vuex'
 import { FlaskConicalIcon } from 'satellite-lucide-icons'
-import { AlertState } from '~/libraries/ui/Alerts'
-import { RootState } from '~/types/store/store'
+import iridium from '~/libraries/Iridium/IridiumManager'
+import { Notification } from '~/libraries/Iridium/notifications/types'
```
Comment (stavares843) [inline components/views/navigation/toolbar/alerts/Alerts.vue:8]: Captura de ecrã 2022-08-16, às 20 23 49


this import is not being used 🔨

**Comment by WanderingHogan** [comment]
@Jekrimo can you delete that lock file and add pnpm-lock.yaml to the gitignore

File: libraries/Iridium/NotificationManager.ts [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  IridiumGetOptions,
+  IridiumSetOptions,
+} from '@satellite-im/iridium'
+import { IridiumManager } from '~/libraries/Iridium/IridiumManager'
+import logger from '~/plugins/local/logger'
+import {
+  EmptyNotification,
+  Notification,
+  NotificationsError,
+  NotificationType,
```
Comment (stavares843) [inline libraries/Iridium/NotificationManager.ts:13]: Captura de ecrã 2022-08-16, às 20 25 50

seems this one is not used 🔨

**Comment by Jekrimo** [comment]
> npx prettier --write

ran it, can we leave the commented out stuff? It'll be uncommented once we get images workin

File: libraries/Iridium/NotificationManager.ts [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+  }
+
+  async deleteNotification(Id: string) {
+    await this.iridium.connector?.set(`/notifications/${Id}`, EmptyNotification)
+  }
+
+  /**
+   * Send a new notification
+   * @method sendNotification
+   * @returns returns the textile response
```
Comment (stavares843) [inline libraries/Iridium/NotificationManager.ts:99]: are we still using textile here? 🔨

Comment (Jekrimo) [reply libraries/Iridium/NotificationManager.ts]: awesome catch!!

**Comment by stavares843** [comment]
sounds good, thanks @Jekrimo 🔨

**Pull request merged**